### ISSUE TITLE
Implement full gradient operation

### ISSUE THREAD

**Issue by radeusgd** [Luna-Tensorflow/Luna-Tensorflow#42]
Current gradient operation can only do differentiation wrt. one scalar parameter, TF allows computing derivatives of multiple parameters or multi-dimensional parameters at once. This will be important for #27.
The behaviour should be analyzed further and should be possible to reproduce in Luna.
(Or maybe we can already differentiate wrt. a tensor and so that's enough of what we need? In this case the only thing to do would be to document that information and add some example using this).

**Comment by radeusgd** [comment]
What's currently available is enough for making backpropagation, but anyway we may want to provide the full functionality (albeit I think under another function and keeping the simple gradient the way it is for simplicity).

**Comment by radeusgd** [comment]
Using Python as reference, it seems that when you provide multiple xs they are just summed, so it doesn't add more functionality, but probably may be faster in some cases?
For multiple ys, the result is an array with the same amount of entries where each entry is a gradient wrt. to corresponding entry in ys.

**Comment by wojtrz13** [comment]
Done in commits: fae92c56de88ce6d30a886a3006e5bef7d004273, db9c470de8cb6fcc2cce2cbd18769367d37c408a, c451e2d4aae9bca27cf0f804163053b6356433c7, bdbb2a3ba739681f45a27ec2b0dc204c08895119
